strawberry/field.py

```diff
@@ -307,45 +310,94 @@ def type_params(self) -> List[TypeVar]:
             return self.type.type_params
         return []
 
+    def resolve_type(
+        self,
+        *,
+        type_definition: Optional[StrawberryObjectDefinition] = None,
+    ) -> Union[  # type: ignore [valid-type]
+        StrawberryType,
+        Type[WithStrawberryObjectDefinition],
+        Literal[UNRESOLVED],
+    ]:
+        # We return UNRESOLVED by default, which means this case will raise a
+        # MissingReturnAnnotationError exception in _check_field_annotations
+        resolved = UNRESOLVED
+        allow_generic = True
```

Comment:
do you think we can remove this allow generic?

We can maybe use a property? but my main issue with this is that I'm not sure how it is used

Reply:
Explaining what is happening here:

If you look at the previous code, `type_annotation.resolve()` got returned regardless, while `base_resolver.type` would force `UNRESOLVED` if it was generic.

I tried handling handling `base_resolver` both ways, as allowing generic to be returned and not, but those broke some tests and I did not get too deep to try to understand why, so I decided to keep the old behaviour in this case, while still trying to solve both when they are specialized.

I can probably add a TODO comment in here

Reply:
let's do that 😊

Reply:
Hrmm, interestingly enough once I refactored the code to not replace the `origin` anymore all tests are now fine D:

Removing the `allow_generic` as it is not required anymore.